packages/babel-plugin/src/utils/export-traversers.tsx

```diff
@@ -0,0 +1,87 @@
+import traverse from '@babel/traverse';
+import type { NodePath } from '@babel/traverse';
+import * as t from '@babel/types';
+
+type Result<T> = {
+  node: t.Node;
+  path: NodePath<T>;
+};
+
+/**
+ * Find the default export of a file and return the export's node and path.
+ *
+ * E.g: `export default 'blue';`
+ * Will return the string literal node for 'blue'.
+ *
+ * E.g: `export default color;`
+ * Will return the identifier node for `color`.
+ *
+ * @param ast File we want to traverse.
+ */
+export const getDefaultExport = (ast: t.File): Result<t.ExportDefaultDeclaration> | undefined => {
+  let result;
+
+  traverse(ast, {
```

Comment:
Not really a problem for the PR as this is the status quo but we'll probably have scale issues unless we remove this stuff per https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#avoid-traversing-the-ast-as-much-as-possible

Reply:
👍 Maybe I should do a `replaceWith` and build an object expression? Is traversing an object expression a lot cheaper? I would also have to add all the exports as properties even if they are not used?

Reply:
`export default` should always be top-level. You be able to manually `find` it in the `program` nodes quite easily instead of `traverse`. Same applies to `getNamedExport`

Reply:
@MonicaOlejniczak I started implementing the loop and remembered we need to get the path for metadata :( Not sure if there is a way to get the path of a node. Should do a `traverse` with enter then do the loop?

Reply:
I'd just leave it as is for now